lib/matplotlib/cbook.py

```diff
@@ -492,17 +504,23 @@ def connect(self, s, func):
         func will be called
         """
         self._func_cid_map.setdefault(s, WeakKeyDictionary())
-        if func in self._func_cid_map[s]:
```

Comment:
Why does this fail for bound methods?  I would have thought that it would fall back to using the memory address as the hash and still work correctly.....